### PR TITLE
CI development workflow RFC

### PR DESCRIPTION
**Pull request by jluk-codi**

### CODE REVIEW

File: source/rfc/ci-development-workflow.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
+CI development workflow
+=======================
+
+CI projects
+-----------
+
+The contrail-project-config repo (hereafter referred to as "c-p-c") contains main Zuul configuration and job definitions that are shared among many projects. Here you'll find:
+- list of projects that can be verified in Zuul
+- for the core projects: which CI jobs are executed by Zuul
+- Zuul workflow definition: what are the
```
Comment (kravvcuCL) [inline source/rfc/ci-development-workflow.rst]: unfinished sentence

File: source/rfc/ci-development-workflow.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+As the contrail-project-config jobs can use secrets, reviews for this
+project will get effective only after a merge. In Zuul this is called a
+"config" or "trusted" project. When you submit a review for c-p-c that
+changes a job, the change will not be used during execution of the tests for
+this review, as it happens for other projects.
+
+
+The contrail-zuul-jobs repo contains other jobs and roles that do not
+require access to secrets and are safe to test during the review process.
+This makes them easier to develop, as one can test the changes before merging
```
Comment (kravvcuCL) [inline source/rfc/ci-development-workflow.rst]: missing . at end of sentence

File: source/rfc/ci-development-workflow.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+TODO (force-merging, fixing errors in c-p-c)
+
+
+Testing changes in contrail-zuul-jobs
+-------------------------------------
+
+The contrail-zuul-jobs repo contains roles and jobs for various projects, so
+it is impractical to execute all the affected jobs in the check pipeline for
+this repo. The only checks that are performed are unit tests and linting of
+the Ansible code. If you're preparing an change that modifies a job for a
```
Comment (kravvcuCL) [inline source/rfc/ci-development-workflow.rst]: 'a' change

**Review by kravvcuCL — DISMISSED** [review]

File: source/rfc/ci-development-workflow.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+Each CI project has a corresponding user group that contains core reviewers 
+- contrail-project-config-core
+- contrail-zuul-jobs-core
+
+Each change to those projects should go through a two-step review process:
+
+#. First core reviewer that accepts the change should place a ``Code-Review +2``
+   vote
+#. Second core reviewer that accepts the change should place an ``Approved +1``
+   vote that will result in a merge (if the CI tests pass)
```
Comment (wurbanski) [inline source/rfc/ci-development-workflow.rst]: I suggest Adding information about preventing merge by giving a own CR-2 vote (if actually possible)

File: source/rfc/ci-development-workflow.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
+CI development workflow
+=======================
+
+CI projects
```
Comment (diabelko) [inline source/rfc/ci-development-workflow.rst]: I think 'zuul-jobs' is missing here. Also, might be worth to move every repo as a separate subparagraph.

File: source/rfc/ci-development-workflow.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+- contrail-zuul-jobs-core
+
+Each change to those projects should go through a two-step review process:
+
+#. First core reviewer that accepts the change should place a ``Code-Review +2``
+   vote
+#. Second core reviewer that accepts the change should place an ``Approved +1``
+   vote that will result in a merge (if the CI tests pass)
+
+Emergency procedures
```
Comment (diabelko) [inline source/rfc/ci-development-workflow.rst]: not sure if this should be a part of the general workflow

File: source/rfc/ci-development-workflow.rst (hunk 5 above)
Comment (diabelko) [reply source/rfc/ci-development-workflow.rst]: contrail-infra as well, considering the workflow is somewhat special there

File: source/rfc/index.rst [hunk 7]
```diff
@@ -2,6 +2,7 @@ Proposals and standards
 =======================
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 1
```
Comment (diabelko) [inline source/rfc/index.rst:5]: nice one ;)

File: source/rfc/ci-development-workflow.rst [hunk 8]
```diff
@@ -0,0 +1,121 @@
+CI development workflow
+=======================
+
+This document describes the rules that control the development of source code responsible for the testing and build of OpenContrail.
+
+Commit message style guide
+--------------------------
+
+|  The commit summary/title line should bo no longer than 50 characters.
```
Comment (wurbanski) [inline source/rfc/ci-development-workflow.rst]: My suggestion: title contains information about what is changed specifically, e.g. "Adds parse-all-the-things job"

Comment (jluk-codi) [reply source/rfc/ci-development-workflow.rst]: Not really defined for me (like "a description of the change should describe what has changed"), but I added something.

**Review by wurbanski — APPROVED** [review]
LGTM

**Pull request merged**